### PR TITLE
Pipeline cache support

### PR DESCRIPTION
**Pull request by kvark**

### CODE REVIEW

File: libportability-gfx/src/impls.rs [hunk 1]
```diff
@@ -1660,23 +1663,28 @@ pub extern "C" fn gfxGetPipelineCacheData(
...
-    _pSrcCaches: *const VkPipelineCache,
+    gpu: VkDevice,
+    dstCache: VkPipelineCache,
+    srcCacheCount: u32,
+    pSrcCaches: *const VkPipelineCache,
 ) -> VkResult {
+    let caches = unsafe {
+        slice::from_raw_parts(pSrcCaches, srcCacheCount as usize)
+    };
+    gpu.device.merge_pipeline_caches(&*dstCache, caches.into_iter().map(|h| h.unbox().unwrap()));
```
Comment (msiglreith) [inline libportability-gfx/src/impls.rs]: We are effectively destroying/consuming the src caches here, right? I don't think that's the intended design of the vulkan specs.

Comment (kvark) [reply libportability-gfx/src/impls.rs]: aaah, where have you been before :)
gotta fix gfx-rs first then

Comment (msiglreith) [reply libportability-gfx/src/impls.rs]: These iterator interfaces are hard to parse :D

**Comment by kvark** [comment]
This is fixed now.
bors r=msiglreith
